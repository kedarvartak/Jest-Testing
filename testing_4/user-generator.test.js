const { createUser } = require('./user-generator');

test('createUser returns a user object', () => {
    const user = createUser('Alice', 30, 'New York');

    // This will create a snapshot file the first time.
    // On subsequent runs, it will compare the new object to the snapshot.
    // This test will fail on the second run because createdAt and id will be different.
    expect(user).toMatchSnapshot();
});

test('createUser returns a consistent user object structure', () => {
    const user = createUser('Bob', 25, 'London');

    // Here's how we fix the problem of dynamic data.
    // We can tell Jest to expect a specific type for certain properties.
    expect(user).toMatchSnapshot({
        createdAt: expect.any(Date),
        id: expect.any(Number),
    });
}); 