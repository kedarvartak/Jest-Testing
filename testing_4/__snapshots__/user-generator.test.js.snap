// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createUser returns a consistent user object structure 1`] = `
{
  "age": 25,
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "location": "London",
  "name": "Bob",
}
`;

exports[`createUser returns a user object 1`] = `
{
  "age": 30,
  "createdAt": 2025-07-24T06:50:28.927Z,
  "id": 274,
  "location": "New York",
  "name": "Alice",
}
`;
